File: 002/Erlang/problem002_tests.erl
-module(problem002_tests).

-include_lib("eunit/include/eunit.hrl").
-import(problem002, [fibSum/1, takeFibs/1]).

fibSum_test_() -> [
	?_assertEqual(0, fibSum(0)),
	?_assertEqual(0, fibSum(1)),
	?_assertEqual(2, fibSum(2)),
	?_assertEqual(10, fibSum(33)),
	?_assertEqual(44, fibSum(34)),
	?_assertEqual(44, fibSum(35))
	].

takeFibs_test_() -> [
	?_assertEqual([], takeFibs(0)),
	?_assertEqual([1], takeFibs(1)),
	?_assertEqual([2, 1], takeFibs(2)),
	?_assertEqual([21, 13, 8, 5, 3, 2, 1], takeFibs(33)),
	?_assertEqual([34, 21, 13, 8, 5, 3, 2, 1], takeFibs(34))
	].
	

File: 002/Python/Problem002Tests.py
#!/usr/bin/env python

import unittest

import problem002

class Problem002Tests(unittest.TestCase):

    def test_fib_sum(self):
        fib_sum = problem002.fib_sum
        self.assertEqual(0, fib_sum(0))
        self.assertEqual(0, fib_sum(1))
        self.assertEqual(2, fib_sum(2))
        self.assertEqual(10, fib_sum(33))
        self.assertEqual(44, fib_sum(34))
        self.assertEqual(44, fib_sum(35))

    def test_fibs_upto(self):
        fibs_upto = problem002.fibs_upto
        self.assertEqual([], fibs_upto(0))
        self.assertEqual([1], fibs_upto(1))
        self.assertEqual([2, 1], fibs_upto(2))
        self.assertEqual([21, 13, 8, 5, 3, 2, 1], fibs_upto(33))
        self.assertEqual([34, 21, 13, 8, 5, 3, 2, 1], fibs_upto(34))
	
unittest.main()
